### ISSUE TITLE
PlatformNotSupportedException on Blazor WebAssembly (not AspNet hosted)

### ISSUE THREAD

**Issue by miggat** [influxdata/influxdb-client-csharp#495]
__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

1. Create a Blazor WebAssembly project using the default template, uncheck ASP.NET Core Hosted, and select .NET6
2. Install InfluxDb.Client 4.11.0
3. Anywhere in the code, try to instantiate a new client:

```
var options = new InfluxDBClientOptions("http://localhost8086")
{
	Token = myToken,
	Timeout = TimeSpan.FromSeconds(10)
};
var client = new InfluxDBClient(options);
```

__Expected behavior:__
Constructor should executed without issues

__Actual behavior:__
Constructor throws a PlatformNotSupportedException 

__Specifications:__
 - Client Version: 4.11.0
 - InfluxDB Version: 2.0.9
 - Platform: Windows 10, Visual Studio 2022
 
I attached a project to reproduce the issue:

[ReproduceInfluxError.zip](https://github.com/influxdata/influxdb-client-csharp/files/11128643/ReproduceInfluxError.zip)

**Comment by alespour** [comment]
This will be fixed in the next release. `InfluxDBClientOptions` has been extended with `HttpClient` property via #528.

InfluxDBClient internally uses `RestSharp.RestClient`. As described in [Blazor support](https://restsharp.dev/usage.html#blazor-support), when used in webassembly, the client has to be instantiated in a specific way due to platform limitations. The above change will allow user to pass `HttpClient` instance, and instantiation of `InfluxDBClient` (`RestCllient` underneath, respectivelly) with such options will work in webassembly. Eg.

```csharp
var options = new InfluxDBClientOptions("http://localhost8086")
{
	Token = myToken,
        HttpClient = new HttpClient(),
	Timeout = TimeSpan.FromSeconds(10)
};
var client = new InfluxDBClient(options);
```

**Comment by alespour** [comment]
Closed via #528
